### ISSUE TITLE
setting `beancountFormatter.binPath` seems to require absolute path

### ISSUE THREAD

**Issue by cookie223** [dongfg/vscode-beancount-formatter#5]
I got the following error if I set the setting `beancountFormatter.binPath` to a relative path `venv/bin/bean-format`.
```
TypeError: Cannot read property '1' of null
```

btw it would be nicer to check the path/output and tell the user what is actually wrong.

Using absolute path seems to work but it is a bit annoying when switching between different work environments.

The `vscode-beancount` plugin can take relative paths as settings for `pythonPath`, `favaPath` etc. The usage of `favaPath` is [here](https://github.com/Lencerf/vscode-beancount/blob/master/src/favaManager.ts#L35). I wonder if the same way can be used for this plugin as well.

**Comment by dongfg** [comment]
I test relative path `venv/bin/bean-format` with venv active, but I cannot reproduce it.
You can test with default value `bean-format`, but don't forget active venv in terminal or auto active in vscode with python extension installed.

**Comment by cookie223** [comment]
In both environments I use, WSL:Ubuntu-18.04 and MacOS Mojave.
`bean-format` or `venv/bin/bean-format` don't work.
 The absolute path (which is different for the two) works. 

The stacktrace:
```
[2020-01-01 11:53:54.639] [exthost] [error] [dongfg.vscode-beancount-formatter] provider FAILED
[2020-01-01 11:53:54.640] [exthost] [error] TypeError: Cannot read property '1' of null
	at Object.provideDocumentFormattingEdits (/home/qchu6/.vscode-server/extensions/dongfg.vscode-beancount-formatter-1.2.0/out/extension.js:39:30)
	at /home/qchu6/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0/out/vs/server/remoteExtensionHostProcess.js:596:205
	at /home/qchu6/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0/out/vs/server/remoteExtensionHostProcess.js:52:287
	at new Promise (<anonymous>)
```

The error message is not the most intuitive but it seems that `child.spawnSync` result is unexpected not handled.

I have python extension set up and it auto activate the venv after I open a python file. I tried "Shift + Alt + F" after that too but still the same error. The `PATH` variable taken in the `env` is probably not updated. 

Can you try not having a `bean-format` in the regular PATH without any venv? I think that's why you couldn't reproduce it.

**Comment by dongfg** [comment]
From the stacktrace, are you using `vscode-server` ? I can reproduce with it.
and `child.spawnSync` result:
```json
{
  "error": {
    "errno": "ENOENT",
    "code": "ENOENT",
    "syscall": "spawnSync <bean-format path>",
    "path": "<bean-format path>",
    "spawnargs": []
  },
  "status": null,
  "signal": null,
  "output": null,
  "pid": 16153,
  "stdout": null,
  "stderr": null
}
```
btw, I shoud display a more friendly message.

**Comment by cookie223** [comment]
The `vscode-server` is probably because I was running it on WSL. On MacOS, using `venv/bin/bean-format` I will get
```
[2020-01-02 14:48:45.337] [exthost] [error] [dongfg.vscode-beancount-formatter] provider FAILED
[2020-01-02 14:48:45.337] [exthost] [error] TypeError: Cannot read property '1' of null
	at Object.provideDocumentFormattingEdits (/Users/qchu6/.vscode/extensions/dongfg.vscode-beancount-formatter-1.2.0/out/extension.js:39:30)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:578:205
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:52:287
	at new Promise (<anonymous>)
```

**Comment by dongfg** [comment]
The direct reason of `Cannot read property '1' of null` is spawnSync's output is null (command not found).
A bug found when `process.cwd()`, using `remote-wsl`, the `process.cwd()` is not workspace root, I have publish a new version(1.2.1) to fix it. I don't know if it fixes your problem. You can try it.
I also add some console log, open `Help -> Toggle Developer Tools` to see.

**Comment by cookie223** [comment]
Confirming fixed in 1.2.1. Thank you so much!
